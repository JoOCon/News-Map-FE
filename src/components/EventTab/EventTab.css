.tab-container {
  z-index: 10;
  position: absolute;
  background-color: #e4e4e4;
  box-shadow: 2px 2px #cdcdcd;
  height: 100vh;
  width: 19rem;
  font-size: 0.6rem;
  overflow-y: auto; }

.tab-img {
  height: 10rem;
  width: 19rem; }

.tab-card {
  border-bottom: 3px solid #d5d5d5;
  padding-bottom: 1rem; }

.tab-card:hover {
  background-color: aliceblue; }

.tab-scroll-container {
  margin: 10rem 0 0 0; }
