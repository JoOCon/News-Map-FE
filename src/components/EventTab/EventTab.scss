.tab-container {
  z-index: 10;
  position: absolute;
  background-color: rgb(228, 228, 228);
  box-shadow: 2px 2px rgb(205, 205, 205);
  height: 100vh;
  width: 19rem;
  font-size: 0.6rem;
}

.tab-img {
  height: 10rem;
  width: 19rem;
}

.tab-card {
  border-top: 3px solid rgb(213, 213, 213);
  padding-bottom: 1rem;
  cursor: pointer;
  filter: grayscale(40%);
}

.tab-card:hover {
  filter: grayscale(0%);
}

.tab-scroll-container {
  margin: 7rem 0 0 0;
  overflow-y: auto;
  height: 90vh;
}
