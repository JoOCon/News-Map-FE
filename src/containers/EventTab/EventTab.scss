.tab-container {
  z-index: 10;
  position: absolute;
  background-color: rgb(170, 170, 170);
  box-shadow: 2px 2px 2px 2px rgba(172, 172, 172, 0.7);
  height: 100vh;
  width: 19rem;
  font-size: 0.6rem;
}

.tab-img {
  height: 10rem;
  width: 15rem;
  border-radius: 1rem;
  align-self: center;
  margin-left: 2rem;
}

.tab-card {
  border-top: 3px solid rgb(213, 213, 213);
  padding-bottom: 1rem;
  cursor: pointer;
  filter: grayscale(40%);
}

.tab-card:hover {
  filter: grayscale(0%);
  background-color: rgb(207, 206, 206);
}

.tab-scroll-container {
  margin: 7rem 0 0 0;
  overflow-y: auto;
  height: 90vh;
  display: flex;
  flex-direction: column;
}

.tab-event-name {
  font-family: caviar;
  letter-spacing: 2px;
  text-align: center;
}

@media only screen and (max-width: 1025px) {
  .tab-scroll-container {
    margin: 0;
    flex-direction: row;
  }

  .tab-img {
    height: 8rem;
    width: 12rem;
  }

  .tab-event-name {
    font-size: 0.7rem;
  }

  .tab-container {
    margin-left: 0;
    z-index: 10;
    width: 100vw;
    height: 13rem;
    font-size: 0.6rem;
    bottom: 0;
  }

  .tab-card {
    border-right: 3px solid rgb(213, 213, 213);
    padding: 0 1rem;
  }
}

@media only screen and (max-width: 1025px) {
  .tab-img {
    height: 5rem;
    width: 8rem;
    margin: 0;
  }

  .tab-container {
    height: 11.5rem;
  }
}
