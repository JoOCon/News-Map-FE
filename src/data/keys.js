export const googleClientId =
  '241779944002-qa42v6cvk3kpu8da0jimdr364vn41t1r.apps.googleusercontent.com';
