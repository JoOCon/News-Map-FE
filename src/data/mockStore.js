import * as user from './mockUser';
import * as events from './mockEvents';

export const mockStore = {
  activeUser: user.mockStateUser,
  userLocation: { latitude: 39.43243, longitude: -104.98434 },
  watchList: events.userWatchlist,
  events: events.eventsResponse
};
