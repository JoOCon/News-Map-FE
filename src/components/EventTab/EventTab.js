import React, { Component } from 'react';
import { connect } from 'react-redux';
import './EventTab.css';

class EventTab extends Component {
  displayEventTab = () => {};

  render() {
    const { events } = this.props;
    console.log(events);
    const x = events.map(event => {
      return <div>{event.name}</div>;
    });
    return <div className="tab-name">{x}</div>;
  }
}

export const mapStateToProps = state => ({
  events: state.events
});

export default connect(mapStateToProps)(EventTab);
