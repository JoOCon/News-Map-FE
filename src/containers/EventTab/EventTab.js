import React, { Component } from 'react';
import { connect } from 'react-redux';
import './EventTab.css';

export class EventTab extends Component {
  render() {
    const { events, showEventInfo, closePopup, handleModalClick } = this.props;
    const eventTab = events.map((event, index) => {
      if (event.name.length > 36) {
        event.name = event.name.slice(0, 36).concat('...');
      }
      return (
        <div
          className="tab-card"
          onMouseEnter={e => showEventInfo(e, event)}
          onMouseLeave={closePopup}
          onClick={event => handleModalClick(event, 'open')}
          key={`tab-${index}`}
        >
          <h1 className="tab-event-name">{event.name}</h1>
          <img alt="event" src={event.img} className="tab-img" />
        </div>
      );
    });
    return (
      <div className="tab-container">
        <div className="tab-scroll-container">{eventTab}</div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  events: state.events
});

export default connect(mapStateToProps)(EventTab);
