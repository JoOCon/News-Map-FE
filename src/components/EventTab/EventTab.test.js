import React from 'react';

import { EventTab, mapStateToProps } from './EventTab';
import '../../data/mockEvents';
import * as events from '../../data/mockEvents';
import { mockStore } from '../../data/mockStore';
import { exact } from 'prop-types';

describe(EventTab, () => {
  let wrapper;
  let mockClose;
  let mockShow;
  let mockHandle;

  beforeEach(() => {
    mockClose = jest.fn();
    mockShow = jest.fn();
    mockHandle = jest.fn();
    wrapper = shallow(
      <EventTab
        showEventInfo={mockShow}
        closePopup={mockClose}
        handleModalClick={mockHandle}
        events={events.eventsResponse}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map to the store properly', () => {
    const expected = mockStore.events;

    const mapped = mapStateToProps(mockStore);

    expect(mapped.events).toEqual(expected);
  });

  it('should call showEventInfo on mouse enter', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper
      .find('.tab-card')
      .first()
      .simulate('mouseEnter', mockEvent, events.mockTargetEventFalse);

    expect(mockShow).toHaveBeenCalledWith(mockEvent, events.eventsResponse[0]);
  });

  it('should call closePopup on mouse leave', () => {
    wrapper
      .find('.tab-card')
      .first()
      .simulate('mouseLeave');

    expect(mockClose).toHaveBeenCalled();
  });

  it('should call handleModalClick on click', () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper
      .find('.tab-card')
      .first()
      .simulate('click', mockEvent);

    expect(mockHandle).toHaveBeenCalledWith(mockEvent, 'open');
  });
});
