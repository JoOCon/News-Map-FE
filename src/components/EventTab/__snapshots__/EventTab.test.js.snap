// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTab should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventTab
    closePopup={[MockFunction]}
    events={
      Array [
        Object {
          "address": "1000 Chopper Circle Denver, CO 80204",
          "date": "Mon, Apr 1, 2019 7:30 PM",
          "distance": 0.75,
          "e_id": "G5vzZ4amDV3z0",
          "favorite": false,
          "img": "https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg",
          "lat": "39.74724",
          "lng": "-105.010166",
          "name": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
          "url": "https://www.altitudetickets.com/events/detail/pink-2019",
          "venue_name": "Pepsi Center",
        },
        Object {
          "address": "1000 Chopper Circle Denver, CO 80204",
          "date": "Tue, Dec 4, 2018 7:30 PM",
          "distance": 0.75,
          "e_id": "G5vzZ4aV5O3CO",
          "favorite": true,
          "img": "https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg",
          "lat": "39.74724",
          "lng": "-105.010166",
          "name": "Childish Gambino",
          "url": "https://www.altitudetickets.com/events/detail/childish-gambino",
          "venue_name": "Pepsi Center",
        },
      ]
    }
    handleModalClick={[MockFunction]}
    showEventInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="tab-scroll-container"
      >
        <div
          className="tab-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[MockFunction]}
        >
          <h1
            className="tab-event-name"
          >
            P!NK: BEAUTIFUL TRAUMA WORLD TOUR
          </h1>
          <img
            alt="event"
            className="tab-img"
            src="https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg"
          />
        </div>
        <div
          className="tab-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[MockFunction]}
        >
          <h1
            className="tab-event-name"
          >
            Childish Gambino
          </h1>
          <img
            alt="event"
            className="tab-img"
            src="https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg"
          />
        </div>
      </div>,
      "className": "tab-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tab-card"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[MockFunction]}
          >
            <h1
              className="tab-event-name"
            >
              P!NK: BEAUTIFUL TRAUMA WORLD TOUR
            </h1>
            <img
              alt="event"
              className="tab-img"
              src="https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg"
            />
          </div>,
          <div
            className="tab-card"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[MockFunction]}
          >
            <h1
              className="tab-event-name"
            >
              Childish Gambino
            </h1>
            <img
              alt="event"
              className="tab-img"
              src="https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg"
            />
          </div>,
        ],
        "className": "tab-scroll-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "tab-0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="tab-event-name"
              >
                P!NK: BEAUTIFUL TRAUMA WORLD TOUR
              </h1>,
              <img
                alt="event"
                className="tab-img"
                src="https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg"
              />,
            ],
            "className": "tab-card",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "onMouseLeave": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
                "className": "tab-event-name",
              },
              "ref": null,
              "rendered": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "event",
                "className": "tab-img",
                "src": "https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "tab-1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="tab-event-name"
              >
                Childish Gambino
              </h1>,
              <img
                alt="event"
                className="tab-img"
                src="https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg"
              />,
            ],
            "className": "tab-card",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "onMouseLeave": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Childish Gambino",
                "className": "tab-event-name",
              },
              "ref": null,
              "rendered": "Childish Gambino",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "event",
                "className": "tab-img",
                "src": "https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tab-scroll-container"
        >
          <div
            className="tab-card"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[MockFunction]}
          >
            <h1
              className="tab-event-name"
            >
              P!NK: BEAUTIFUL TRAUMA WORLD TOUR
            </h1>
            <img
              alt="event"
              className="tab-img"
              src="https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg"
            />
          </div>
          <div
            className="tab-card"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[MockFunction]}
          >
            <h1
              className="tab-event-name"
            >
              Childish Gambino
            </h1>
            <img
              alt="event"
              className="tab-img"
              src="https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg"
            />
          </div>
        </div>,
        "className": "tab-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="tab-card"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[MockFunction]}
            >
              <h1
                className="tab-event-name"
              >
                P!NK: BEAUTIFUL TRAUMA WORLD TOUR
              </h1>
              <img
                alt="event"
                className="tab-img"
                src="https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg"
              />
            </div>,
            <div
              className="tab-card"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[MockFunction]}
            >
              <h1
                className="tab-event-name"
              >
                Childish Gambino
              </h1>
              <img
                alt="event"
                className="tab-img"
                src="https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg"
              />
            </div>,
          ],
          "className": "tab-scroll-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "tab-0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="tab-event-name"
                >
                  P!NK: BEAUTIFUL TRAUMA WORLD TOUR
                </h1>,
                <img
                  alt="event"
                  className="tab-img"
                  src="https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg"
                />,
              ],
              "className": "tab-card",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
                  "className": "tab-event-name",
                },
                "ref": null,
                "rendered": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "event",
                  "className": "tab-img",
                  "src": "https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "tab-1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="tab-event-name"
                >
                  Childish Gambino
                </h1>,
                <img
                  alt="event"
                  className="tab-img"
                  src="https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg"
                />,
              ],
              "className": "tab-card",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Childish Gambino",
                  "className": "tab-event-name",
                },
                "ref": null,
                "rendered": "Childish Gambino",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "event",
                  "className": "tab-img",
                  "src": "https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
